src/swap/types.ts

```diff
@@ -3,47 +3,82 @@ import { Token } from '../token/types';
 
+/**
+ * Note: exported as public Type
+ */
 export type Fee = {
-  amount: string;
-  baseAsset: Token;
-  percentage: string;
+  amount: string; // The amount of the fee
+  baseAsset: Token; // The base asset for the fee
+  percentage: string; // The percentage of the fee
 };
 
+export type GetQuoteAPIParams = {
+  from: AddressOrETH | ''; // The source address or 'ETH' for Ethereum
+  to: AddressOrETH | ''; // The destination address or 'ETH' for Ethereum
+  amount: string; // The amount to be swapped
+  amountReference?: string; // The reference amount for the swap
+};
+
+/**
+ * Note: exported as public Type
+ */
+export type GetQuoteParams = {
+  from: Token; // The source token for the swap
+  to: Token; // The destination token for the swap
+  amount: string; // The amount to be swapped
+  amountReference?: string; // The reference amount for the swap
+  amountInDecimals?: boolean; // Whether the amount is in decimals
+};
+
+/**
+ * Note: exported as public Type
+ */
+export type GetQuoteResponse = Quote | SwapError;
+
+/**
+ * Note: exported as public Type
+ */
 export type Quote = {
-  amountReference: string;
-  fromAmount: string;
-  fromAsset: Token;
-  highPriceImpact: boolean;
-  priceImpact: string;
-  slippage: string;
-  toAmount: string;
-  toAsset: Token;
-  warning?: QuoteWarning;
+  amountReference: string; // The reference amount for the quote
+  from: Token; // The source token for the swap
+  fromAmount: string; // The amount of the source token
+  highPriceImpact: boolean; // Whether the price impact is high
+  priceImpact: string; // The price impact of the swap
+  slippage: string; // The slippage of the swap
+  to: Token; // The destination token for the swap
+  toAmount: string; // The amount of the destination token
+  warning?: QuoteWarning; // The warning associated with the quote
 };
 
+/**
+ * Note: exported as public Type
+ */
 export type QuoteWarning = {
-  description?: string;
-  message?: string;
-  type?: string;
+  description?: string; // The description of the warning
+  message?: string; // The message of the warning
+  type?: string; // The type of the warning
 };
 
-export type Trade = {
-  approveTx?: Transaction;
-  chainId: string;
-  fee: Fee;
-  tx: Transaction;
+/**
+ * Note: exported as public Type
+ */
+export type SwapError = {
+  code: number; // The error code
+  error: string; // The error message
 };
 
-export type Transaction = {
-  data: string;
-  from: string;
-  gas: string;
-  gasPrice: string;
-  to: string;
-  value: string;
+export type Trade = {
+  approveTx?: Transaction; // The approval transaction
+  chainId: string; // The chain ID
+  fee: Fee; // The fee for the trade
+  tx: Transaction; // The trade transaction
 };
 
-export type TrendingToken = Token & {
-  numOfBuys: number;
-  numOfSells: number;
+export type Transaction = {
+  data: string; // The transaction data
+  from: string; // The sender address
+  gas: string; // The gas limit
```

Comment:
Open qustion, should this be `gasLimit` as we have `gasPrice`?

Reply:
Or this is just a standard Viem and other have already? cc @0xAlec @cpcramer @kyhyco

Reply:
will iterate on this once I integrate `getTrade` - this type is unused for `getQuote`